test/testsuites/cscpackagetests.go

```diff
@@ -0,0 +1,123 @@
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	f := test.Global
+
+	// Run tests.
+	if err := cscWithNonExistingPackage(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// cscWithNonExistingPackage is a test case that creates a CatalogSourceConfig
+// with a package that doesn't exist and ensures that child resources are not 
+// created.
+func cscWithNonExistingPackage(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// nonExistingPackageCSCName is the name of the catalogsourceconfig that points
+	// to a non-existing Package
+	nonExistingPackageCSCName string = "non-existing-package-operators"
+
+	// packageName is the non-existing package
+	packageName string = "non-existing-package"
+
+	// Create a new catalogsourceconfig with a non-existing Package
+	nonExistingPackageCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      nonExistingPackageCSCName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        packageName,
+		}}
+	err = helpers.createRuntimeObject(f, ctx, nonExistingPackageCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with a non-existing package
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = helpers.WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
+	expectedMessage := fmt.Sprintf("Still resolving package(s) - %v. Please make sure these are valid packages.", packageName)
+	// Check that the catalogsourceconfig with an non-existing package eventually reaches the
+	// configuring phase with the expected message
+	err = wait.Poll(retryInterval, timeout, func() (bool, error) {
+		err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+		if err != nil {
+			return false, err
+		}
+		if resultCatalogSourceConfig.Status.CurrentPhase.Name == expectedPhase &&
+			resultCatalogSourceConfig.Status.CurrentPhase.Message == expectedMessage {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("CatalogSourceConfig never reached expected phase/message, expected %v/%v", expectedPhase, expectedMessage)
+	}
+
+	catalogSourceName := nonExistingPackageCSCName
+	serviceName := nonExistingPackageCSCName
+	deploymentName := nonExistingPackageCSCName
+
+	// Check catalog source was not created.
+	resultCatalogSource := &olm.CatalogSource{}
+	err = helpers.WatiForNotFound(t, resultCatalogSource, namespace, catalogSourceName)
```

Comment:
Did you actually try running this after you made this change? Please test locally before asking us to review.